spark/src/main/scala/ai/chronon/spark/Analyzer.scala

```diff
@@ -251,16 +251,34 @@ class Analyzer(tableUtils: TableUtils,
 
   def analyzeJoin(joinConf: api.Join,
                   enableHitter: Boolean = false,
-                  validationAssert: Boolean = false): (Map[String, DataType], ListBuffer[AggregationMetadata]) = {
+                  metadataExtraction: Boolean = false,
+                  validationAssert: Boolean = false
+                 ): (Map[String, DataType], ListBuffer[AggregationMetadata]) = {
     val name = "joins/" + joinConf.metaData.name
     logger.info(s"""|Running join analysis for $name ...""".stripMargin)
     // run SQL environment setups such as UDFs and JARs
     joinConf.setups.foreach(tableUtils.sql)
 
-    val leftDf = JoinUtils.leftDf(joinConf, range, tableUtils, allowEmpty = true).get
-    val analysis = if (enableHitter) analyze(leftDf, joinConf.leftKeyCols, joinConf.left.table) else ""
-    val leftSchema: Map[String, DataType] =
-      leftDf.schema.fields.map(field => (field.name, SparkConversions.toChrononType(field.name, field.dataType))).toMap
+    val (analysis, leftSchema) = if (enableHitter) {
+      val leftDf = JoinUtils.leftDf(joinConf, range, tableUtils, allowEmpty = true).get
+      val analysis = analyze(leftDf, joinConf.leftKeyCols, joinConf.left.table)
+      val leftSchema: Map[String, DataType] =
+        leftDf.schema.fields.map(field => (field.name, SparkConversions.toChrononType(field.name, field.dataType))).toMap
+      (analysis, leftSchema)
+    } else if (metadataExtraction){
```

Comment:
I think we can simplify this logic. Basically we only do the analysis for heavy hitters. Otherwise I think we can use the same schema extraction logic for all flows.